idx/memory/memory.go

```diff
@@ -484,49 +534,46 @@ func (m *UnpartitionedMemoryIdx) indexTags(def *schema.MetricDefinition) {
 // a return value of "false" means there was an error and the deindexing was
 // unsuccessful, "true" means the indexing was at least partially or completely
 // successful
-func (m *UnpartitionedMemoryIdx) deindexTags(tags TagIndex, def *schema.MetricDefinition) bool {
-	for _, tag := range def.Tags {
-		tagSplits := strings.SplitN(tag, "=", 2)
-		if len(tagSplits) < 2 {
-			// should never happen because every tag in the index
-			// must have a valid format
-			invalidTag.Inc()
-			log.Errorf("memory-idx: Tag %q of id %q has an invalid format", tag, def.Id)
-			continue
-		}
-
-		tagName := tagSplits[0]
-		tagValue := tagSplits[1]
-		tags.delTagId(tagName, tagValue, def.Id)
+func (m *UnpartitionedMemoryIdx) deindexTags(tags TagIndex, def *idx.MetricDefinition) bool {
+	for _, tag := range def.Tags.KeyValues {
+		tags.delTagId(tag.Key, tag.Value, def.Id)
 	}
 
-	tags.delTagId("name", def.Name, def.Id)
+	nameKey, _ := idx.IdxIntern.GetPtrFromByte([]byte("name"))
+	nameValue, _ := idx.IdxIntern.GetPtrFromByte([]byte(def.Name.String()))
```

Comment:
shouldn't those errors be handled to prevent that invalid values get passed into `delTagId()`?

Reply:
Yes